php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1011 @@
+        b.optional(
+          f.newTuple92(
+            b.token(PHPPunctuator.EQU), 
+            STATIC_SCALAR())))
+      );      
+  }
+
+  /**
+   * [ END ] Declaration
+   */
+
+  /**
+   * [ START ] Statement
+   */
+
+  public StatementTree TOP_STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.TOP_STATEMENT)
+        .is(b.firstOf(
+//            CLASS_DECLARATION(),
+            FUNCTION_DECLARATION(),
+//            INTERFACE_DECLARATION(),
+            NAMESPACE_STATEMENT(),
+            USE_STATEMENT(),
+//            CONSTANT_DECLARATION(),
+            STATEMENT()
+        ));
+  }
+
+  public NamespaceStatementTree NAMESPACE_STATEMENT() {
+    return b.<NamespaceStatementTree>nonterminal(PHPLexicalGrammar.NAMESPACE_STATEMENT)
+        .is(b.firstOf(
+           f.namespaceStatement(
+               b.token(PHPKeyword.NAMESPACE),
+               NAMESPACE_NAME(),
+               EOS()
+           ),
+           f.blockNamespaceStatement(
+               b.token(PHPKeyword.NAMESPACE),
+               b.optional(NAMESPACE_NAME()),
+               b.token(LCURLYBRACE),
+               b.zeroOrMore(TOP_STATEMENT()),
+               b.token(RCURLYBRACE)
+           )
+        ));
+  }
+
+  public UseStatementTree USE_STATEMENT() {
+    return b.<UseStatementTree>nonterminal(PHPLexicalGrammar.USE_STATEMENT).is(
+        f.useStatement(
+            b.token(PHPKeyword.USE),
+            b.optional(
+                b.firstOf(
+                    b.token(PHPKeyword.CONST),
+                    b.token(PHPKeyword.FUNCTION))),
+            USE_DECLARATION(),
+            b.zeroOrMore(f.newTuple90(b.token(PHPPunctuator.COMMA), USE_DECLARATION())),
+            EOS())
+    );
+  }
+
+  public StatementTree STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.STATEMENT)
+        .is(b.firstOf(
+            BLOCK(),
+            THROW_STATEMENT(),
+            IF_STATEMENT(),
+            WHILE_STATEMENT(),
+            DO_WHILE_STATEMENT(),
+            FOREACH_STATEMENT(),
+            FOR_STATEMENT(),
+            SWITCH_STATEMENT(),
+            BREAK_STATEMENT(),
+            CONTINUE_STATEMENT(),
+            RETURN_STATEMENT(),
+            EMPTY_STATEMENT(),
+            YIELD_STATEMENT(),
+            GLOBAL_STATEMENT(),
+            STATIC_STATEMENT(),
+            TRY_STATEMENT(),
+            DECLARE_STATEMENT(),
+            GOTO_STATEMENT(),
+            INLINE_HTML(),
+            UNSET_VARIABLE_STATEMENT(),
+            EXPRESSION_STATEMENT(),
+            LABEL()
+        ));
+  }
+
+  public StaticStatementTree STATIC_STATEMENT() {
+    return b.<StaticStatementTree>nonterminal(PHPLexicalGrammar.STATIC_STATEMENT)
+        .is(f.staticStatement(
+            b.token(STATIC),
+            STATIC_VAR(),
+            b.zeroOrMore(f.newTuple22(b.token(COMMA), STATIC_VAR())),
+            EOS()
+        ));
+  }
+
+  public VariableDeclarationTree STATIC_VAR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)